testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -5118,107 +5118,7 @@ fn check_block_heights() {
-            &naka_conf,
-            &sender_addr,
-            contract3_name,
-            "get-heights",
-            vec![],
-        );
-        let heights3 = heights3_value.expect_tuple().unwrap();
-        info!("Heights from Clarity 3: {}", heights3);
-
-        let bbh1 = heights1
-            .get("burn-block-height")
-            .unwrap()
-            .clone()
-            .expect_u128()
-            .unwrap();
-        let bbh3 = heights3
-            .get("burn-block-height")
-            .unwrap()
-            .clone()
-            .expect_u128()
-            .unwrap();
-        assert_eq!(bbh1, bbh3, "Burn block heights should match");
-        last_burn_block_height = bbh1;
-
-        let bh1 = heights1
-            .get("block-height")
-            .unwrap()
-            .clone()
-            .expect_u128()
-            .unwrap();
-        let bh3 = heights3
-            .get("tenure-height")
-            .unwrap()
-            .clone()
-            .expect_u128()
-            .unwrap();
-        assert_eq!(
-            bh1, bh3,
-            "Clarity 2 block-height should match Clarity 3 tenure-height"
-        );
-        assert_eq!(
-            bh1,
-            last_tenure_height + 1,
-            "Tenure height should have incremented"
-        );
-        last_tenure_height = bh1;
-
-        let sbh = heights3
-            .get("stacks-block-height")
-            .unwrap()
-            .clone()
-            .expect_u128()
-            .unwrap();
-        let expected_height = if tenure_ix == 0 {
-            // tenure 0 will include an interim block at this point because of the contract publish
-            //  txs
-            last_stacks_block_height + 2
-        } else {
-            last_stacks_block_height + 1
-        };
-        assert_eq!(
-            sbh, expected_height,
-            "Stacks block heights should have incremented"
-        );
-        last_stacks_block_height = sbh;
-
-        // mine the interim blocks
-        for interim_block_ix in 0..inter_blocks_per_tenure {
-            info!("Mining interim block {interim_block_ix}");
-            let blocks_processed_before = coord_channel
-                .lock()
-                .expect("Mutex poisoned")
-                .get_stacks_blocks_processed();
-            // submit a tx so that the miner will mine an extra block
-            let transfer_tx =
-                make_stacks_transfer(&sender_sk, sender_nonce, send_fee, &recipient, send_amt);
-            sender_nonce += 1;
-            submit_tx(&http_origin, &transfer_tx);
-
-            loop {
-                let blocks_processed = coord_channel
-                    .lock()
-                    .expect("Mutex poisoned")
-                    .get_stacks_blocks_processed();
-                if blocks_processed > blocks_processed_before {
-                    break;
-                }
-                thread::sleep(Duration::from_millis(100));
-            }
-
+        wait_for(30, || {
```

Comment:
yes it seemed to . Without this, it would randomly be off by 1. 30 seconds is prob more than required, byt it doesn't really hurt either except to cause the test to fail in 30 vs 20 vs 10 seconds.